tests/__init__.py

```diff
@@ -0,0 +1,112 @@
+import os
+import unittest
+
+import responses
+
+import roboflow
+from roboflow.config import API_URL
+
+ROBOFLOW_API_KEY = "my-test-app"
+WORKSPACE_NAME = "my-workspace"
+PROJECT_NAME = "test-project"
+PROJECT_VERSION = "1"
+
+
+def make_orderer():
+    order = {}
+
+    def ordered(f):
+        order[f.__name__] = len(order)
+        return f
+
+    def compare(a, b):
+        if a not in order or b not in order:
+            return 1
+
+        return [1, -1][order[a] < order[b]]
+
+    return ordered, compare
+
+
+ordered, compare = make_orderer()
+
+unittest.defaultTestLoader.sortTestMethodsUsing = compare
+
+
+class RoboflowTest(unittest.TestCase):
+
+    def setUp(self):
+        super(RoboflowTest, self).setUp()
+        responses.start()
+
+        # Check key
+        responses.add(
+            responses.POST,
+            f"{API_URL}/?api_key={ROBOFLOW_API_KEY}",
+            json={
+                "welcome": "Welcome to the Roboflow API.",
+                "instructions": "You are successfully authenticated.",
+                "docs": "https://docs.roboflow.com",
+                "workspace": WORKSPACE_NAME
+            },
+            status=200
```

Comment:
it would be nice to abstract this mocking a bit further and only mock when necessary... but this seemed like an easy to start in order to stop hitting the API during automated tests.

i also have `WORKSPACE_NAME` and `PROJECT_NAME` constants in the JSON responses because i was initially reading from `.env` vars... but we don't need those anymore, so i may just hardcode the strings. 

it'd be nice to eventually move these responses into easily replaceable json blobs that are read from a file or something.

Reply:
Yep I could definitely see the file of JSON blobs - and we add/edit one when we add/edit a Rest API route

This is awesome!